src/index.ts

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+/**
+ * Singleton Class that will keep track of the active function
+ * 
+ * We need the singleton instance of this to keep track of a function running
+ * in order to add it as an action on a list of actions in Depend class
+ * 
+ * We use the observe method to intercept each property of the state
+ */
+class JsUpdatr {
```

Comment:
You could split the responsibility and set this code to its own file. 
`JsUpdatr.ts` (suggestion: since it's a class, you should start with capital letter)
`Depend.ts`
and another file called 
`index.ts` where you can do:
```
export * from './JsUpdatr'
export * from './Watcher'

Reply:
more to come.. after my meetings lol